### PR TITLE
Setup CircleCI

### PR DESCRIPTION
**Pull request by markusjura**
Sets up CircleCI to run tests for each PR and master commit. Later, we could also add a workflow step to publish to maven central, if necessary.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Scala CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/sample-config/ for more details
+#
+version: 2
+jobs:
+  build:
+    docker:
+    - image: hseeberger/scala-sbt:8u181_2.12.6_1.2.1
```
Comment (hseeberger) [inline .circleci/config.yml]: Nice ;-)

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    working_directory: ~/repo
+
+    environment:
+      SBT_OPTS: "-Xms256m -Xmx2g -XX:+UseConcMarkSweepGC -XX:ReservedCodeCacheSize=256m -XX:+CMSClassUnloadingEnabled -Djava.awt.headless=true"
```
Comment (hseeberger) [inline .circleci/config.yml]: I'd love to see a comment explaining these settings.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    steps:
+    - checkout
+    # Download and cache dependencies
+    - restore_cache:
+        keys:
+        - sbt-v1-deps-{{ checksum "build.sbt" }}
+        # fallback to using the latest cache if no exact match is found
+        - sbt-v1-deps
+    - run:
+        name: Running SBT pre-compilation
```
Comment (hseeberger) [inline .circleci/config.yml]: Why is thie pre-compilation. It's real compilation.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  build:
+    docker:
+    - image: hseeberger/scala-sbt:8u181_2.12.6_1.2.1
+
+    working_directory: ~/repo
+
+    environment:
+      # Setting the JVM minimum memory allocation to 256 MB and maximum allocation to 2 GB.
+      # java.awt.headless=true: Disabling GUI features because streamee don't use them
+      SBT_OPTS: "-Xms256m -Xmx2g -Djava.awt.headless=true"
```
Comment (markusjura) [inline .circleci/config.yml]: Removed unnecessary options. They can be used for larger projects but for this OSS library they are not necessary.

**Comment by hseeberger** [comment]
Thanks!

**Pull request merged**